saleor/graphql/order/bulk_mutations/order_bulk_create.py

```diff
@@ -543,27 +703,113 @@ def validate_order_input(cls, order_input, errors: List[OrderBulkError]):
             try:
                 validate_storefront_url(redirect_url)
             except ValidationError as err:
-                errors.append(
+                order.errors.append(
                     OrderBulkError(
                         message=f"Invalid redirect url: {err.message}.",
-                        field="redirect_url",
+                        path="redirect_url",
                         code=OrderBulkCreateErrorCode.INVALID,
                     )
                 )
 
         weight = order_input.get("weight")
         if weight and weight.value < 0:
-            errors.append(
+            order.errors.append(
+                OrderBulkError(
+                    message="Order can't have negative weight.",
+                    path="weight",
+                    code=OrderBulkCreateErrorCode.INVALID,
+                )
+            )
+
+        if number := order_input.get("number"):
+            lookup_key = f"Order.number.{number}"
+            if object_storage.get(lookup_key):
+                order.errors.append(
+                    OrderBulkError(
+                        message=f"Order with number: {number} already exists.",
+                        path="number",
+                        code=OrderBulkCreateErrorCode.UNIQUE,
+                    )
+                )
+                order.is_critical_error = True
+
+    @classmethod
+    def validate_order_status(cls, status: str, order: OrderBulkClass):
+        total_order_quantity = order.total_order_quantity
+        total_fulfillment_quantity = order.total_fulfillment_quantity
+
+        is_invalid = False
+        if total_fulfillment_quantity == 0 and status in [
+            OrderStatus.PARTIALLY_FULFILLED,
+            OrderStatus.FULFILLED,
+        ]:
+            is_invalid = True
+        if (
+            total_fulfillment_quantity > 0
+            and (total_order_quantity - total_fulfillment_quantity) > 0
+            and status in [OrderStatus.FULFILLED, OrderStatus.UNFULFILLED]
+        ):
+            is_invalid = True
+        if total_order_quantity == total_fulfillment_quantity and status in [
+            OrderStatus.PARTIALLY_FULFILLED,
+            OrderStatus.UNFULFILLED,
+        ]:
+            is_invalid = True
+
+        if is_invalid:
+            order.errors.append(
                 OrderBulkError(
-                    message="Product can't have negative weight.",
-                    field="weight",
+                    message="Invalid order status.",
+                    path="status",
                     code=OrderBulkCreateErrorCode.INVALID,
                 )
             )
 
-        # TODO validate status (wait for fulfillments)
-        # TODO validate number (?)
-        return errors
+    @classmethod
+    def validate_order_numbers(cls, orders: List[OrderBulkClass]):
+        numbers = []
+        for order in orders:
+            if order.order and order.order.number in numbers:
```

Comment:
`order.order` looks strange, maybe to differentiate `Order` instance and the `OrderBulkClass` instance it should be `order_data.order`?

Reply:
good points, thanks